task/rhacs-image-scan/3.71/rhacs-image-scan.yaml

```diff
@@ -49,28 +52,119 @@ spec:
 
         When set to "true", skip verifying the TLS certs of the Central endpoint and registry.
       default: "false"
+  volumes:
+    - name: token-vol
+      projected:
+        sources:
+          - serviceAccountToken:
+              audience: api
+              expirationSeconds: 3600
+              path: token
+  workspaces:
+    - name: roxctl-config
+      description: The folder where the roxctl config should be stored.
+      mountPath: /roxctl-config
+  stepTemplate:
+    env:
+      - name: HOME
+        value: /tekton/home
+      - name: INSECURE
+        value: $(params.insecure-skip-tls-verify)
+      - name: ROX_CONFIG_DIR
+        value: $(workspaces.roxctl-config.path)
+      - name: ROX_CENTRAL_ENDPOINT
+        valueFrom:
+          secretKeyRef:
+            key: rox_central_endpoint
+            name: $(params.rox_central_endpoint)
   steps:
+    - name: extract-serviceaccount-token
```

Comment:
I suppose this, at a latter point, would be a `roxctl central m2m login` call only, right?

With all the environment variables set, it should be simply:
```bash
roxctl central m2m exchange --token=$(cat /service-account-token)
```

Reply:
Technically, this should be a `roxctl central m2m exchange` call.
I haven't found a clean way of propagating the system account token into an environment variable to use with the roxctl image. I found something about creating a secret wrapping around the service account token, but then the token content does not contain the correct issuer information.
I think the exchange command would require a `--token-file` flag as alternative to the `--token` one.